sentry-spring/src/main/java/io/sentry/spring/reactive/SentryReactiveUserProvider.java

```diff
@@ -0,0 +1,16 @@
+package io.sentry.spring.reactive;
+
+import io.sentry.protocol.User;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+
+/**
+ * Provides user information that's set on {@link io.sentry.SentryEvent}.
+ *
+ * <p>Out of the box Spring integration configures single {@link SentryReactiveUserProvider}.
+ */
+@FunctionalInterface
+public interface SentryReactiveUserProvider {
```

Comment:
could this interface be merged with https://github.com/getsentry/sentry-java/blob/main/sentry-spring/src/main/java/io/sentry/spring/SentryUserProvider.java ?
not sure if it'd make sense though, this one requires a `ServerHttpRequest`

Reply:
Not sure it can given that it comes from `org.springframework.http.server.reactive.ServerHttpRequest`

The `reactive` there might not exist at runtime given the `compileOnly` import

Reply:
@marandaneto as bruno says, if we merge these two interfaces there will be problems when using SentryUserProvider in spring MVC. In Webflux we can not extract the User information from the current Thread as the Thread model is very different, thus we need extract it from the thread independent Request object.